### ISSUE TITLE
Run server bundle in spawned process

### ISSUE THREAD

**Issue by enten** [enten/tension#4]
Currently, the server bundle is running in the same process as its compiler.

So, we need to take care about cache invalidation, http listener lifecycle (close connections and listener) and error catching (to not crash the compiler process).

**It's will be better to spawn a child process to run the server bundle.**

This approach offer these benefits:

* don't manage http listener lifecycle (listener will close when on process exit)
* the entry point isn't longer obliged to export http listener as default
* not needed to manage cache invalidation

**Comment by ctrlplusb** [comment]
Very interesting! Be careful with not explicitly closing the listeners though. You may want to do some POC around this first.

**Comment by enten** [comment]
@ctrlplusb 

I hesitated to open this issue on react-universally. Glad to see that this topic attracted your attention.

> Be careful with not explicitly closing the listeners though.

You're right!

We all encountered an issue when a process is over but the http listener is still running. And to fix that issue we kill the PID of the phantom process. 

**This is what we need to do on compiler `done` event: [kill the spawned process](https://nodejs.org/api/child_process.html#child_process_child_kill_signal).**

I made some experimentation this morning. I will continue tonight and maybe push changes if the result is suitable.
